action_sync/src/file.rs

```diff
@@ -0,0 +1,263 @@
+use anyhow::anyhow;
+use anyhow::Result;
+use bytes::{Buf, BufMut, BytesMut};
+use std::fmt::Formatter;
+use std::fs::{File, OpenOptions};
+use std::io::{BufReader, Read, Seek, SeekFrom, Write};
+use std::path::Path;
+use tezos_context::channel::ContextActionMessage;
+
+use crypto::hash::HashType;
+use serde::{Deserialize, Serialize};
+
+const HEADER_LEN: usize = 44;
+const BLOCK_HASH_HEADER_LEN: usize = 32;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Block {
+    pub block_level: u32,
+    pub block_hash_hex: String,
+    pub block_hash: [u8; BLOCK_HASH_HEADER_LEN],
+    pub predecessor: [u8; BLOCK_HASH_HEADER_LEN],
+}
+
+impl Block {
+    pub fn new(block_level: u32, raw_block_hash: Vec<u8>, raw_predecessor: Vec<u8>) -> Self {
```

Comment:
someone on security channel said that he is working on removeing Vec<u8> as hash representation instead array are used. It might be good idea to follow that approach with new development :thinking:

Reply:
Yeah, i would make some of the code better. is there a pull request for it?